### PR TITLE
patch ghost advance tests

### PR DESCRIPTION
**Pull request by PhilipDeegan**

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
remove the deprecated init tests now, or wait till these changes are reviewed?

File: pyphare/pyphare/pharesee/hierarchy.py [hunk 1]
```diff
@@ -1076,7 +1076,7 @@ def hierarchy_from(simulator=None, qty= None, pop = "", h5_filename=None, time=N
 def merge_particles(hierarchy):
 
     for time, patch_levels in hierarchy.time_hier.items():
-        for ilvl, plvl in hierarchy.patch_levels.items():
+        for ilvl, plvl in patch_levels.items():
```
Comment (nicolasaunai) [inline pyphare/pyphare/pharesee/hierarchy.py:1079]: maybe we should rather use the levels() method? (if we deprecate patch_levels soonish)

File: tests/simulator/test_advance.py [hunk 2]
```diff
@@ -143,6 +168,258 @@ def vthz(x):
 
 
 
+    def _test_overlaped_fields_are_equal(self, time_step, time_step_nbr, datahier):
```
Comment (nicolasaunai) [inline tests/simulator/test_advance.py:171]: that's a copy/paste from initialization test, do we want that?

**Review by nicolasaunai — COMMENTED** [review]
I would think that for those methods that exist in the initialization test that are repeated here, those here also test the initialization (t=0) no? thus I would remove them in the init test. If not I would like to understand why, and how to avoid possible duplication (of not so trivial code btw)

File: tests/simulator/test_advance.py (hunk 2 above)
Comment (PhilipDeegan) [reply tests/simulator/test_advance.py:171]: copy/paste/change
this one has advances, and I would say deprecates the initialization version, which I haven't removed but can as an addendum to this PR

Comment (nicolasaunai) [reply tests/simulator/test_advance.py:171]: yes I would remove all initialization tests that are redundant with these advance tests

Comment (PhilipDeegan) [reply tests/simulator/test_advance.py:171]: commit/pushed/building

**Review by nicolasaunai — APPROVED** [review]
squash & merge?

**Pull request merged**